server/market/types/types.go

```diff
@@ -0,0 +1,151 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/decred/dcrdex/server/asset"
+)
+
+// MarketInfo specified a market that the Archiver must support.
+type MarketInfo struct {
+	Name    string
+	Base    uint32
+	Quote   uint32
+	LotSize uint64
+}
+
+func marketName(base, quote string) string {
+	return base + "_" + quote
+}
+
+func MarketName(base, quote uint32) (string, error) {
+	baseSymbol := asset.BipIDSymbol(base)
+	if baseSymbol == "" {
+		return "", fmt.Errorf("base asset %d not found", base)
+	}
+	baseSymbol = strings.ToLower(baseSymbol)
+
+	quoteSymbol := asset.BipIDSymbol(quote)
+	if quoteSymbol == "" {
+		return "", fmt.Errorf("quote asset %d not found", quote)
+	}
+	quoteSymbol = strings.ToLower(quoteSymbol)
+
+	return marketName(baseSymbol, quoteSymbol), nil
+}
+
+func NewMarketInfo(base, quote uint32, lotSize uint64) (*MarketInfo, error) {
+	name, err := MarketName(base, quote)
+	if err != nil {
+		return nil, err
+	}
+	return &MarketInfo{
+		Name:    name,
+		Base:    base,
+		Quote:   quote,
+		LotSize: lotSize,
+	}, nil
+}
+
+func NewMarketInfoFromSymbols(base, quote string, lotSize uint64) (*MarketInfo, error) {
+	baseID, found := asset.BipSymbolID(strings.ToLower(base))
+	if !found {
+		return nil, fmt.Errorf(`base asset symbol "%s" unrecognized`, base)
+	}
+
+	quoteID, found := asset.BipSymbolID(strings.ToLower(quote))
+	if !found {
+		return nil, fmt.Errorf(`quote asset symbol "%s" unrecognized`, quote)
+	}
+
+	return &MarketInfo{
+		Name:    marketName(base, quote),
+		Base:    baseID,
+		Quote:   quoteID,
+		LotSize: lotSize,
+	}, nil
+}
+
+// OrderStatus indicates how an order is presently being processed, or the final
+// state of the order if processing is completed.
+type OrderStatus uint16
+
+const (
+	// OrderStatusUnknown is a sentinel value to be used when the status of an
+	// order cannot be determined.
+	OrderStatusUnknown OrderStatus = iota
+
+	// There are two general classes of orders: ACTIVE and TERMINAL. Orders with
+	// one of the ACTIVE order statuses that follow are likely to be updated.
+
+	// OrderStatusPending is for active orders that have been received and
+	// validated, but not processed by the epoch order matcher.
+	OrderStatusPending
+	// OrderStatusMatched is for active orders that have been matched with other
+	// orders, but for which an atomic swap has not been initiated.
+	OrderStatusMatched
+	// OrderStatusSwapping is for active orders that have begun the atomic swap
+	// process. Specifically, this is when the first swap initialization
+	// transaction has been broadcast by a client.
+	OrderStatusSwapping
+	// OrderStatusBooked is for active unmatched orders that have been put on
+	// the order book (standing time in force).
+	OrderStatusBooked
```

Comment:
I was approaching the status column of the orders tables a bit differently, but I see the issues you're describing.  Although, describing the filled state explicit (unfilled, partial, or filled) overlaps with the filled amount column (quantity vs. filled amounts describe the fill state).

`Rejected` (meaning received but not passed order validation), would never enter the table at all, so `Epoch` would effectively be the same as `Pending`.

As far as the `Matched` and `Swapping` statuses, I see how that is problematic since an order can be matched with multiple orders, and it can be involved in swaps while simultaneously (1) back on the books with a reduced remaining amount or (2) matched once again with a new taker.  Perhaps a state to describe an order that is "unbooked" but not yet executed by cancellation or a complete fill.  As one example,

`Pending` (validated, but still in queue) -> `Booked` -> **`Unbooked`** (see swaps or matches tables for details) -> `Booked` (with reduced amount) -> `Executed` or `Canceled`

Reply:
Actually, I think I completely agree with you now, at least regarding the "levels".  I don't think having `Unbooked` as a state makes much sense anymore since the order amount can simply be reduced when and if it gets matched.

Reply:
As for market orders, they would need a status like Unbooked or OffBook while they are being executed.  A market order that has been matched and sent to swap doesn't fit into any of these:

```go
Rejected
Epoch
Booked
Executed
Canceled
```

Reply:
Why can't it be `Executed`?

Reply:
I'm guess because I'm thinking of `Executed` as meaning a successful trade, although you're keeping the scope of order status much more limited (with match and swap status as a downstream issue).  I appreciate that, but I'm not yet convinced there won't be ambiguous cases that will require extra effort to get the required information that the archivist consumer needs.

I'll go through with the changes to the proposed states, except for Rejected which as I mentioned won't hit the DB.  If I'm wrong about that too, please call me out, but my thinking is that it's a very easy way to attack the DEX by submitting invalid orders (wrong lot size or other issue).